vizro-core/src/vizro/actions/export_data_class_action.py

```diff
@@ -0,0 +1,110 @@
+import importlib
+from typing import Any, Dict, List, Literal
+
+from dash import Output, State, ctx, dcc
+
+from vizro.actions._filter_action import _filter
+from vizro.actions.filter_interaction_action import filter_interaction
+from vizro.managers import model_manager
+from vizro.models.types import CapturedActionCallable
+
+
+class ExportDataClassAction(CapturedActionCallable):
+    def __init__(self, *args, **kwargs):
+        self._args = args
+        self._kwargs = kwargs
+        # Fake initialization - to let other actions see that this one exists.
+        super().__init__(*args, **kwargs)
+
+    def _post_init(self):
+        """Post initialization is called in the vm.Action build phase, and it is used to validate and calculate the
+        properties of the CapturedActionCallable. With this, we can validate the properties and raise errors before
+        the action is built. Also, "input"/"output"/"components" properties and "pure_function" can use these validated
+        and the calculated arguments.
+        """
+        self._page_id = model_manager._get_model_page_id(model_id=self._action_id)
+
+        # Validate and calculate "targets"
+        targets = self._kwargs.get("targets")
+        if targets:
+            for target in targets:
+                if target not in model_manager:
+                    raise ValueError(f"Component '{target}' does not exist on the page '{self._page_id}'.")
+        else:
+            targets = model_manager._get_page_model_ids_with_figure(page_id=self._page_id)
+        self._kwargs["targets"] = self.targets = targets
```

Comment:
Not sure I understand what's happening with the `_kwargs` stuff here, please could you explain?

Reply:
Now this line looks like this:
`self._arguments["targets"] = self.targets = targets`

- `targets` - represents new validated and calculated action's targets
- `self._arguments["targets"] = targets` - overwrites the `pure_function` input argument. Since the pure_function is a staticmethod (and I'm pretty sure it should remain), new calculated targets have to be propagated through the `self._arguments`.
- `self.targets = targets` - Optionally, some of CapturedActionCallable attributes are also calculated in the _post_init phase. In this case, `self.targets` is created calculated so it can be easily reused inside the `outputs and `components` calculations.